src/interpolation_caches.jl

```diff
@@ -366,6 +366,78 @@ function ConstantInterpolation(
         cache_parameters, assume_linear_t)
 end
 
+"""
+    SmoothedConstantInterpolation(u, t; d_max = Inf, extrapolate = false,
+        cache_parameters = false, assume_linear_t = 1e-2)
+
+It is a method for interpolating constantly with forward fill, with smoothing around the
+value transitions to make the curve continuously differentiable while the integral never
+drifts far from the integral of constant interpolation. In this interpolation type u[end] is ignored.
+
+## Arguments
+
+  - `u`: data points.
+  - `t`: time points.
+
+## Keyword Arguments
+
+  - `d_max`: the maximum distance in `t` from the data points the smoothing is allowed to reach.
```

Comment:
For users, this might not be very clear. Can you add an explanation like this image in https://github.com/SciML/DataInterpolations.jl/issues/364#issuecomment-2495610505